library_management_system_downloader/create_api_request.py

```diff
@@ -0,0 +1,51 @@
+import backoff
+from ratelimit import rate_limited
+import requests
+import logging
+
+
+class ErrorWithAPI(Exception):
+    pass
+
+
+# From the backoff documentation (https://pypi.python.org/pypi/backoff),
+# set maximum number of tries on a failed download:
+@backoff.on_exception(
+        backoff.expo,
+        requests.exceptions.RequestException,
+        max_tries=8)
+# From the ratelimit documentation (https://pypi.python.org/pypi/ratelimit),
+# set the rate limit (in seconds) for API calls:
+@rate_limited(0.2)
```

Comment:
To make this more clear, can the comment say "max one query per five seconds"?

Reply:
Sure, done in f8cf46c.
To confirm, though, did you mean "max five queries per one second"? (That's what I added in the above-referenced commit, and it seems to be correct from testing the code).

Reply:
From looking at the [source code](https://github.com/tomasbasham/ratelimit/blob/1fe9ff9c18a97ff3e39731fd5d88f4e42a419310/ratelimit/__init__.py), I think the current implementation is saying run at most 0.2 times per second, which gets "clamped" to at most once per second. Documentation is poor, but I think you need to do:

```python
@rate_limited(period=1, every=5)
```

To say that the call can be made at most once within a five second period.

Reply:
If you want "max five queries per one second" I think you want `@rate_limited(5)` which i @rate_limited(period=5, every=1)

Reply:
I confirmed the two-argument behavior you suggested just now with a test, and have updated the comment and syntax (which now uses explicit arguments, and the second (`every`) argument, in f81cd152a.

The [pypi](https://pypi.python.org/pypi/ratelimit) page doesn't mention the second argument, which is why I missed it!